### PR TITLE
Render texture fixes for iOS

### PR DESCRIPTION
**Pull request by SmilyOrg**
Built on and need #311 and #313 first. Look at this after those have landed (they will drop down the diff count below a bit). Render textures should work on iOS with these fixes.

### CODE REVIEW

**Comment by bengarney** [comment]
Needs a very sexy rebase.

File: loom/common/core/assert.h [hunk 1]
```diff
@@ -49,20 +55,26 @@ void loom_fireAssertCallback();
...
 #else
 
+#define lmSafeAssert lmSafeCheck
+#define lmAssert lmCheck
+
+#endif
+
 // Use lmSafeAssert when we are in modules that might not be able to handle malloc() and strcpy()
-#define lmSafeAssert(condition, errmsg) \
+#define lmSafeCheck(condition, errmsg) \
```
Comment (bengarney) [inline loom/common/core/assert.h]: Check vs. assert is super handy but needs to be documented well in this header.

File: loom/graphics/gfxVectorRenderer.cpp [hunk 2]
```diff
@@ -1,544 +1,546 @@
...
+}
+
+void VectorRenderer::beginFrame()
+{
+    nvgBeginFrame(nvg, frameWidth, frameHeight, 1);
+
+
+
+    /*
+    nvgBeginPath(nvg);
```
Comment (bengarney) [inline loom/graphics/gfxVectorRenderer.cpp]: Dead code?

File: loom/graphics/gfxVectorRenderer.cpp [hunk 3]
```diff
@@ -1,544 +1,546 @@
...
+
+NVGcontext *nvg = NULL;
+static int font;
+
+utHashTable<utHashedString, utString> VectorTextFormat::loadedFonts;
+int VectorRenderer::frameWidth = 0;
+int VectorRenderer::frameHeight = 0;
+uint8_t VectorRenderer::quality = VectorRenderer::QUALITY_ANTIALIAS | VectorRenderer::QUALITY_STENCIL_STROKES;
+
+//*
```
Comment (bengarney) [inline loom/graphics/gfxVectorRenderer.cpp]: Weird comments?

File: loom/graphics/gfxVectorRenderer.cpp (hunk 2 above)
Comment (SmilyOrg) [inline loom/graphics/gfxVectorRenderer.cpp]: Oops, looks like I committed weird whitespace at some point. That file is messy, but for another PR I think.

File: loom/graphics/gfxVectorRenderer.cpp (hunk 3 above)
Comment (SmilyOrg) [inline loom/graphics/gfxVectorRenderer.cpp]: Same as above.

**Comment by SmilyOrg** [comment]
Lets hear about this one!

**Pull request merged**